### PR TITLE
Cleanup wpgu for PR

### PR DESCRIPTION
**Pull request by ollisco**

### CODE REVIEW

**Comment by mathiasmagnusson** [comment]
Almost everything in `ide/src/main.rs` should be moved out from there.

Also, the rendering has a long way to go.

**Comment by mathiasmagnusson** [comment]
I made a new crate for stuff that everything will depend on (such as `Transform`) so that when you compile the physics crate, you won't have to also compile the graphics crate with all it's dependencies (but both should use the same `Transform`).

I named it `common`, but I don't like the name. Perhaps it should be called `math(s)` since that's all it contains (for now at least)? Or maybe there is a better way to structure it?

File: ide/src/main.rs [hunk 1]
```diff
@@ -7,5 +9,11 @@ mod window;
 fn main() {
     env_logger::init();
 
-    Editor::new().unwrap().run();
+    let (event_loop, mut editor) = Editor::new().unwrap();
```
Comment (ollisco) [inline ide/src/main.rs:12]: Do we really want to unwrap here?

Comment (mathiasmagnusson) [reply ide/src/main.rs:12]: There isn't much else to do except log the error, but the unwrap should to that.

**Review by mathiasmagnusson — APPROVED** [review]
I think this is done enough to merge into the other branch.

It would however be nice to clean up `Editor` but i don't really know how.

And if someone has a better idea about what to do with the `common` crate, please suggest!

**Pull request merged**